pallets/lbp/src/lib.rs

```diff
@@ -658,7 +666,18 @@ pub mod pallet {
 
 			Self::deposit_event(Event::LiquidityRemoved(pool_id, asset_a, asset_b, amount_a, amount_b));
 
-			Ok(())
+			let amount_a_perbill: Balance = transfer_to_xyk * amount_a;
+			let amount_b_perbill: Balance = transfer_to_xyk * amount_b;
+			// test final values because multiplication of small non-zero values can result in zero value
+			if amount_a_perbill.is_zero() || amount_b_perbill.is_zero() {
+				Ok(())
+			} else {
+				ensure!(
+					pool_data.final_weight == MAX_WEIGHT / 2,
+					Error::<T>::InvalidFinalWeights
+				);
+				T::OnRemoveLiquidity::on_end_of_lbp(who, asset_a, asset_b, amount_a_perbill, amount_b_perbill)
```

Comment:
```suggestion
			let amount_a_to_pool: Balance = transfer_to_xyk * amount_a;
			let amount_b_to_pool: Balance = transfer_to_xyk * amount_b;
			// test final values because multiplication of small non-zero values can result in zero value
			if amount_a_to_pool.is_zero() || amount_b_to_pool.is_zero() {
				Ok(())
			} else {
				ensure!(
					pool_data.final_weight == MAX_WEIGHT / 2,
					Error::<T>::InvalidFinalWeights
				);
				T::OnRemoveLiquidity::on_end_of_lbp(who, asset_a, asset_b, amount_a_to_pool, amount_b_to_pool)
```
nitpick: they're not really perbills :-D